Added dash footer component

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->

#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Dash footer added for improving UI of Dashboard related components.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-158-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->
![dash-footer](https://user-images.githubusercontent.com/17106489/60052780-2c16b780-96a4-11e9-869c-ad8494662755.png)

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

lunayach commented:
@sanketbansal for all these PR's, since the Surge is not working currently, screenshots may help us to see the visual changes that you have made.

sanketbansal commented:
@lunayach @Shekharrajak @RishabhJain2018 @galipremsagar  please review

lunayach commented:
@sanketbansal resolve the conflicts.

lunayach commented:
@sanketbansal I think that this is also not reflected. You need to call the component selector somewhere.

sanketbansal committed:
Added relevant styles

Styles added into the base.scss and dash-footer component

sanketbansal commented:
It is being used by the #141 #149 #143

lunayach commented on src/app/components/nav/dash-footer/dash-footer.component.html:
```diff
@@ -0,0 +1,39 @@
+            <a class="github-button" href="https://github.com/Cloud-CV/EvalAI" data-icon="octicon-star" data-size="large" data-show-count="true" aria-label="Star Cloud-CV/EvalAI on GitHub">Star</a>
+          </li>
+          <li>
+            <!-- Place this tag where you want the button to render. -->
```
No need of these.

sanketbansal replied:
done

lunayach commented on src/app/components/nav/dash-footer/dash-footer.component.ts:
```diff
@@ -0,0 +1,24 @@
+
+  ngOnInit() {
+    this.year = new Date().getFullYear();
+    const js = this.document.createElement('script');
```
Remove comments and console logs.

sanketbansal replied:
done

sanketbansal committed:
Changes in dash footer component

Removed logs and comments from dash footer component

lunayach reviewed (approved):
@RishabhJain2018 we can also merge this now. We can scrutinize the reflection in UI when reviewing #141.

RishabhJain2018 commented:
@sanketbansal Please resolve the conflicts here.

sanketbansal commented:
Done

RishabhJain2018 commented:
@sanketbansal I think you forgot to push the changes.

sanketbansal commented:
> @sanketbansal I think you forgot to push the changes.

Yeah sorry Done now

Shekharrajak commented on src/app/components/nav/dash-footer/dash-footer.component.ts:
```diff
@@ -0,0 +1,22 @@
+    this.year = new Date().getFullYear();
+    const js = this.document.createElement('script');
+    js.src = 'https://buttons.github.io/buttons.js';
+    this.document.getElementsByTagName('head')[0].appendChild(js);
```
We should avoid using plain javascript in our Angular application. Use typescript and Angular features.

Shekharrajak replied:
@sanketbansal  so you want to add buttons.js to use class `github-button` , right ?

Shekharrajak replied:
You can simply add script tag in html file. But I don't think you should  import whole js file just to get github icon. Isn't we are already using font-awesome that has github stars and fork icons?

sanketbansal replied:
it's not only github icons actually it is used to fetch number of stars and forks on the github repository also.

sanketbansal replied:
> Just want to let you know: If you want to manipulate any DOM element (the similar thing that you were doing -- accessing any tag) Angular have `ElementRef `, `ViewRef`, .. reference types to pick any DOM element and modify it. There is `Renderer` to create new elements as well.

Thanks for letting me know. I tried that but I can not get access to head tag do not know why?
Maybe because it's not a component specific element. what do you think?

Shekharrajak reviewed (commented):
I am not seeing the footer as the screenshot attached. Can you check in https://pr-158-evalai.surge.sh/

Shekharrajak commented:
I didn't understand why did you use plain javascript instead of Angular typescript code.

sanketbansal commented:
> I didn't understand why did you use plain javascript instead of Angular typescript code.

I did not find any other method to insert script tag in the head tag.
As I am not appending inside the component-specific element.

sanketbansal commented:
@Shekharrajak is there a way to do it?

Shekharrajak commented:
> @Shekharrajak is there a way to do it?

Commented https://github.com/Cloud-CV/EvalAI-ngx/pull/158#discussion_r305577735

sanketbansal committed:
Removed script tag injection

Added script tag in index.html file
Removed javascript to append script tag into head element

sanketbansal commented:
@Shekharrajak is it okay now

Shekharrajak commented:
Just want to let you know: If you want to manipulate any DOM element (the similar thing that you were doing -- accessing any tag) Angular have `ElementRef `, `ViewRef`, .. reference types to pick any DOM element and modify it. There is `Renderer` to create new elements as well.

RishabhJain2018 committed:
Merge branch 'master' into Added-Dash-Footer
